### PR TITLE
feat : Tiptap integration

### PR DESCRIPTION
**Pull request by Palanikannan1437**
# Summary

This PR focuses on transitioning from Remirror to Tiptap, implementing new features, and fixing bugs. It includes the addition of bubble menu support, slash command support, improved file structure with prosemirror plugins, image upload functionality, and integration of Tiptap with all Remirror instances.

https://github.com/makeplane/plane/assets/73993394/51b95fb8-86f3-4c4e-9169-96865f3ca7f7

# Changes in Detail

## Remirror to Tiptap Transition
The transition from Remirror to Tiptap involved
 - Migrating styles and Ref behaviours to play well with `react-hook-form`
 - Fixing underlying prosemirror state to avoid conflicts
 - Migration wasn't painful for existing data as we now use `html` instead of Prosemirror JSON!

## Bubble Menu Support 
Allowing users to access various formatting options easily with:
 1. Changing list behaviour between todos(tasklists) /numbered/bullet lists
image
 
 2. Toggle code blocks with ```syntax highlighting```
 3. Add hyper links
image
 
 4. Basic marks bold, italic, underline, strikethrough

## Slash Command Support
Enabling users to quickly insert nodes like:
1. Creating Headings and text (h1,h2,h3,p)
2. Different types of lists (todo, bullet, ordered, unordered)
3. Divider
4. Add code snippets with syntax highlighting
5. Upload Images from their File system

## Saving with Debounce Logic
Debounce logic was added to save user input after a 500ms delay when the user stops typing. This change improved the saving process, which previously relied on the editor's blur state that could be very unreliable.

https://github.com/makeplane/plane/assets/73993394/9a6500ac-3892-430a-8f0f-8c119bab112e

## Image Uploads and File Structure Improvements
1. Image upload functionality was implemented, allowing users to upload and `reorder images` within the editor. 
2. Code structure was improved by using native Prosemirror Plugin to implement image uploads. 
3. Delete Image function was implemented to clear clutter in backend S3 file storage like so:
    - keep track of “previous state” of the editor and the “current state”
    - Traverse through both states and compare changes
    - Assign unique ids to all necessary nodes to compare changes while traversing in successive edits
    - Check the node type being undefined and being an `image` so that I can trigger a function to delete it on the backend as well

https://github.com/makeplane/plane/assets/73993394/61296ca0-f124-468d-b7dd-3a1718e54fd3

## Tiptap Integration with Various Components

- Issue Description
- Comments
- User Activity Feeds
- GPT Assistant Modals
- Plane Pages

### Additional Context

The PR also includes various improvements and bug fixes, such as resolving conflicting keyboard shortcuts supporting `markdown` shortcuts, and implementing a "Saved" state for image uploads with placeholder while uploading..

### CODE REVIEW

File: apps/app/components/core/modals/gpt-assistant-modal.tsx [hunk 1]
```diff
@@ -32,17 +32,17 @@ type FormData = {
...
-});
-
-import { IRemirrorRichTextEditor } from "components/rich-text-editor";
-
-const WrappedRemirrorRichTextEditor = forwardRef<IRemirrorRichTextEditor, IRemirrorRichTextEditor>(
-  (props, ref) => <RemirrorRichTextEditor {...props} forwardedRef={ref} />
-);
-
-WrappedRemirrorRichTextEditor.displayName = "WrappedRemirrorRichTextEditor";
+// const RemirrorRichTextEditor = dynamic(() => import("components/rich-text-editor"), {
```
Comment (sriramveeraghanta) [inline apps/app/components/core/modals/gpt-assistant-modal.tsx:35]: Remove this commented code and clear it up, once tiptap editor is working fine.

File: apps/app/components/issues/EditorBubbleMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
+import { BubbleMenu, BubbleMenuProps } from "@tiptap/react";
```
Comment (sriramveeraghanta) [inline apps/app/components/issues/EditorBubbleMenu.tsx:1]: Rename this component file name as the rest of the components with small letters.

Comment (sriramveeraghanta) [inline apps/app/components/issues/EditorBubbleMenu.tsx:1]: Rename this component file name as the rest of the components with small letters. 

Create a new folder in the components folder with the name `tiptap-editor` and place everything related to the editor in that folder.

**Review by sriramveeraghanta — CHANGES_REQUESTED** [review]

File: apps/app/components/issues/EditorBubbleMenu.tsx (hunk 2 above)
Comment (Palanikannan1437) [reply apps/app/components/issues/EditorBubbleMenu.tsx]: Done 🙌

File: apps/app/components/core/modals/gpt-assistant-modal.tsx (hunk 1 above)
Comment (Palanikannan1437) [reply apps/app/components/core/modals/gpt-assistant-modal.tsx:35]: For sure!

**Pull request closed without merging**